src/supplemental.js

```diff
@@ -12,6 +14,10 @@ function calcSupplementalDamage(types, damageArray, vals, {remainHP = 1.0, expec
                 if (remainHP < supplemental.threshold)
                     continue;
                 /* FALLTHROUGH */
+            case "boss_debuff_based":
+                /* FALLTHROUGH */
+            case "djeeta_buff_based":
+                /* FALLTHROUGH */
```

Comment:
This FALLTHROUGH chain may not work as you expected
boss_debuff_based will run all  boss_debuff_based, on_critical and other
or require ensure double type check in the case
case "djeeta_buff_based": if (type was "djeeta_buff_based") { // to avoid anoter type

Reply:
I see now, it currently nothing to do in other fall through cases.

if/else is flexible for this use?

- if (hp_based and remainHP) continue // only filtering
- if (third_hit) ...
- else ... hp_based also add damage here
- no need to care unknown case since it filtered by types.includes

Reply:
I changed their type from "other" to specific (boss_debuff_based/etc) to avoid introducing more values/variables. these are supposed to be same as "other". (and as you mentioned. they're already filtered by include, or it should not get any unexpected values.)